### PR TITLE
Update React to 16.8 to use hooks

### PR DESCRIPTION
**Pull request by stereobooster**
**What**:

**Why**:

**How**:

**Checklist**:

* [ ] Documentation
* [ ] Tests
* [ ] Ready to be merged 
* [ ] Added myself to contributors table

### CODE REVIEW

File: src/components/IdealImage/index.js [hunk 1]
```diff
@@ -97,86 +97,40 @@ const defaultGetIcon = state => {
...
+  const [state, setState] = useState({
+    loadState: initial,
+    connection: nativeConnection
+      ? {
+          downlink: navigator.connection.downlink, // megabits per second
+          rtt: navigator.connection.rtt, // ms
+          effectiveType: navigator.connection.effectiveType, // 'slow-2g', '2g', '3g', or '4g'
+        }
+      : null,
+    onLine: true,
```
Comment (stereobooster) [inline src/components/IdealImage/index.js]: [`useOnLineStatus`](https://stereobooster.github.io/react-hooks/?search=useOnLineStatus)

File: src/components/IdealImage/index.js [hunk 2]
```diff
@@ -185,167 +139,209 @@ export default class IdealImage extends Component {
...
-        return
-      case initial:
-      case error:
-        this.load(true)
-        return
-      default:
-        throw new Error(`Wrong state: ${loadState}`)
-    }
-  }
+    updateOnlineStatus = () => setState({onLine: navigator.onLine})
```
Comment (stereobooster) [inline src/components/IdealImage/index.js]: [`useOnLineStatus`](https://stereobooster.github.io/react-hooks/?search=useOnLineStatus)

File: src/components/Media/index.js [hunk 3]
```diff
@@ -1,165 +1,166 @@
...
-          height={props.height}
-        />
-      </noscript>
-    ) : null
-  }
+const Media = props => {
+  const dimensionElement = useRef(null)
+
+  useEffect(() => {
+    if (props.onDimensions && dimensionElement.current)
```
Comment (stereobooster) [inline src/components/Media/index.js]: [`useSize`](https://stereobooster.github.io/react-hooks/?search=size)

File: src/components/IdealImage/index.js [hunk 4]
```diff
@@ -185,167 +139,209 @@ export default class IdealImage extends Component {
...
+    const {loadState, url} = state
     if (ssr || loaded === loadState || loading === loadState) return
-    this.loadStateChange(loading, userTriggered)
+    loadStateChange(loading, userTriggered)
 
-    const {threshold} = this.props
-    const loader =
-      this.props.loader === 'xhr' ? xhrLoader(url) : imageLoader(url)
+    const {threshold} = props
+    const loader = props.loader === 'xhr' ? xhrLoader(url) : imageLoader(url)
```
Comment (stereobooster) [inline src/components/IdealImage/index.js]: [`useFetch`](https://stereobooster.github.io/react-hooks/?search=fetch)

File: src/components/IdealImage/index.js [hunk 5]
```diff
@@ -185,167 +139,209 @@ export default class IdealImage extends Component {
...
+        return
+      default:
+        throw new Error(`Wrong state: ${loadState}`)
     }
   }
 
-  onEnter = () => {
-    if (this.state.inViewport) return
-    this.setState({inViewport: true})
+  const onEnter = () => {
```
Comment (stereobooster) [inline src/components/IdealImage/index.js]: [`useIntersectionObserver`](https://stereobooster.github.io/react-hooks/?search=intersection)

**Comment by revanth0212** [comment]
Can we break this PR into just updating `react` and move code changes into different PR? Just a suggestion, but if that's not the case, this is good to be merged.

**Pull request closed without merging**